### PR TITLE
Iteration2 develop reporting

### PR DESCRIPTION
**Pull request by mudassarsyed**
This pull request contains inclusion of reporting along with m*n execution

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    stage('Checkout') {
+        sh "git checkout 'iteration2_develop_reporting'"
+    }
+
+
+    stage('Build') {
+
+
+		echo "Hello World"
```
Comment (anirudha13) [inline Jenkinsfile]: Remove

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		sh 'chmod +x gradlew'
+    	browserstack('mudassardemo'){
```
Comment (anirudha13) [inline Jenkinsfile]: Please remove credentials

File: build.gradle [hunk 3]
```diff
@@ -5,12 +5,19 @@
...
+    id "com.github.spacialcircumstances.gradle-cucumber-reporting" version "0.1.22"
 }
 
 repositories {
     mavenLocal()
     mavenCentral()
     google()
+    maven {
+        url = uri('https://repo.maven.apache.org/maven2/')
+    }
```
Comment (anirudha13) [inline build.gradle]: Isn't this the same as mavenCentral?

File: src/test/java/com/browserstack/examples/core/WebDriverFactory.java [hunk 4]
```diff
@@ -151,8 +152,12 @@ private WebDriver createRemoteWebDriver(Platform platform, String testName, Map<
...
         platformCapabilities.setCapability("name", testName);
         platformCapabilities.setCapability("project", commonCapabilities.getProject());
-        platformCapabilities.setCapability("build", createBuildName(commonCapabilities.getBuildPrefix()));
-
+        if (StringUtils.isNotEmpty(buildName)){
+            platformCapabilities.setCapability("build", buildName);
+        }
+        else {
+            platformCapabilities.setCapability("build", createBuildName(commonCapabilities.getBuildPrefix()));
+        }
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/core/WebDriverFactory.java]: Add all this logic into the **createBuildName** function.

File: src/test/java/com/browserstack/examples/core/WebDriverFactory.java [hunk 5]
```diff
@@ -161,15 +166,16 @@ private WebDriver createRemoteWebDriver(Platform platform, String testName, Map<
...
+        //}
         String accessKey = remoteDriverConfig.getAccessKey();
-        if (StringUtils.isNoneEmpty(System.getenv(BROWSERSTACK_ACCESS_KEY))) {
+        //if (StringUtils.isNotEmpty(System.getenv(BROWSERSTACK_ACCESS_KEY))) {
             accessKey = System.getenv(BROWSERSTACK_ACCESS_KEY);
-        }
+        //}
         platformCapabilities.setCapability("browserstack.user", user);
         platformCapabilities.setCapability("browserstack.key", accessKey);
+        System.out.println(user+":"+accessKey);
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/core/WebDriverFactory.java]: Please cleanup.

**Review by anirudha13 — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (mudassarsyed) [reply Jenkinsfile]: Removed echo mesage

File: build.gradle (hunk 3 above)
Comment (mudassarsyed) [reply build.gradle]: yes, updated

File: Jenkinsfile (hunk 2 above)
Comment (mudassarsyed) [reply Jenkinsfile]: used credentials id, referred here: https://www.browserstack.com/docs/automate/selenium/jenkins#using-environment-variables

File: src/test/java/com/browserstack/examples/core/WebDriverFactory.java (hunk 5 above)
Comment (mudassarsyed) [reply src/test/java/com/browserstack/examples/core/WebDriverFactory.java]: done

**Review by anirudha13 — APPROVED** [review]

**Pull request merged**